icu4j/main/classes/core/src/com/ibm/icu/number/NumberFormatterImpl.java

```diff
@@ -50,21 +59,21 @@ public NumberFormatterImpl(MacroProps macros) {
     /**
      * Builds and evaluates an "unsafe" MicroPropsGenerator, which is cheaper but can be used only once.
      */
-    public static int formatStatic(
+    public static MicroProps formatStatic(
             MacroProps macros,
             DecimalQuantity inValue,
             FormattedStringBuilder outString) {
         MicroProps micros = preProcessUnsafe(macros, inValue);
```

Comment:
unitMeasure test currently triggers NullPopinterException in this line:

        if (micros.integerWidth.maxInt == -1) {

Call stack starting at this line (which is at line 66 at the current branch head):

```
java.lang.NullPointerException
at com.ibm.icu.number.NumberFormatterImpl.preProcessUnsafe(NumberFormatterImpl.java:120)
at com.ibm.icu.number.NumberFormatterImpl.formatStatic(NumberFormatterImpl.java:66)
at com.ibm.icu.number.LocalizedNumberFormatter.formatImpl(LocalizedNumberFormatter.java:155)
at com.ibm.icu.number.LocalizedNumberFormatter.format(LocalizedNumberFormatter.java:129)
at com.ibm.icu.number.LocalizedNumberFormatter.format(LocalizedNumberFormatter.java:83)
at com.ibm.icu.dev.test.number.NumberFormatterApiTest.assertFormatSingle(NumberFormatterApiTest.java:3806)
at com.ibm.icu.dev.test.number.NumberFormatterApiTest.unitMeasure(NumberFormatterApiTest.java:649)
```

To be determined: where integerWidth is supposed to be set, and why that was skipped. The unit test in question (:649):

        // TODO(icu-units#35): skeleton generation.
        assertFormatSingle(
                "Mixed unit",
                null,
                "unit/yard-and-foot-and-inch",
                NumberFormatter.with()
                        .unit(MeasureUnit.forIdentifier("yard-and-foot-and-inch")),
                new ULocale("en-US"),
                3.65,
                "3 yd, 1 ft, 11.4 in");

integerWidth gets set up in macrosToMicroGenerator:

        // Integer width
        if (macros.integerWidth != null) {
            micros.integerWidth = macros.integerWidth;
        } else {
            micros.integerWidth = IntegerWidth.DEFAULT;
        }

Which means the problem we have is that this "original micros" is getting lost, instead of cloned as the initial MicroProps instance to build on.

Reply:
Fixed in a60c08230b